model/cat/pit_segments/structures.smithy

```diff
@@ -17,12 +23,47 @@ list PitIds{
 }
 
 @input
-structure CatPitSegments_Input{
+structure CatPitSegments_Input with [CatPitSegments_QueryParams] {
     @httpPayload
-    pit_id: CatPitSegments_BodyParams
+    content: CatPitSegments_BodyParams
+}
+
+@input
+structure CatAllPitSegments_Input with [CatPitSegments_QueryParams] {
 }
 
 @output
 structure CatPitSegments_Output {
-    content: CatPitSegment
+    @httpPayload
+    content: CatPitSegmentsRecords
+}
+
+@output
+structure CatAllPitSegments_Output {
+    @httpPayload
+    content: CatPitSegmentsRecords
+}
+
+list CatPitSegmentsRecords {
+    member: CatPitSegmentsRecord
+}
+
+structure CatPitSegmentsRecord {
+    index: String,
+    shard: String,
+    prirep: String,
+    ip: String,
+    segment: String,
+    generation: String,
+    @jsonName("docs.count")
+    docs_count: String,
+    @jsonName("docs.deleted")
+    docs_deleted: String,
+    size: String,
+    @jsonName("size.memory")
```

Comment:
I'll have to keep this in mind when backfilling the request and response bodies. Thanks :)